unitTests/magicbox/SuggestionsManagerTest.ts

```diff
@@ -411,19 +415,61 @@ export function SuggestionsManagerTest() {
             return Utils.resolveAfter(previewsPromisesWaitTimes[suggestionId], previewsBySuggestion[suggestionId]);
           }
 
+          const delayBeforePopulatePreviews = 150;
+          function bindUpdateResultPreviewsManagerOptionsEvent() {
+            $$(env.root).on(
+              ResultPreviewsManagerEvents.updateResultPreviewsManagerOptions,
+              (_, args: IUpdateResultPreviewsManagerOptionsEventArgs) => {
+                args.delayBeforePopulate = delayBeforePopulatePreviews;
+              }
+            );
+          }
+
           let populateSpy: jasmine.Spy;
+          let waitForPreviewsPopulateEvent: () => Promise<any>;
           function bindPopulateEvent() {
```

Comment:
This function looks really complicated. Are you sure we need it?

Reply:
Unfortunately, it does seem like we need it to test what happens between previews being queried and rendered. Doing a lone `tick` seemed to cause issues. I agree that this is pretty complicated and I would like to simplify it, but at that point I had played with it for a while and I only caused more issues.